### PR TITLE
Add NBT Data to Profile Viewer Tabs and XP Bars

### PR DESCRIPTION
**Pull request by walkerselby**
Add NBT Data Support for Resources in Tabs
Add NBT Data Support for Resources in XP Bars
Optimize renderTabs function
Optimize mouseClicked Function
Optimize ProfileViewerPage Enum

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/GuiProfileViewer.java [hunk 1]
```diff
@@ -539,16 +574,12 @@ public void printStackTrace() {
...
 	private void renderTabs(boolean renderPressed) {
-		int ignoredTabs = 0;
-		for (int i = 0; i < ProfileViewerPage.values().length; i++) {
-			ProfileViewerPage page = ProfileViewerPage.values()[i];
-			if (page.stack == null) {
-				ignoredTabs++;
-				continue;
-			}
+		final List<ProfileViewerPage> pages = ProfileViewerPage.getPagesWithItem();
+		for (int index = 0; pages.size()>index; index++){
```
Comment (NopoTheGamer) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/GuiProfileViewer.java:578]:
```suggestion
		for (int index = 0; pages.size() > index; index++){
```
:slightly_smiling_face:

**Review by romangraef — CHANGES_REQUESTED** [review]
This PR looks fine in general, but there are a LOT of problems with formatting, please see our .editorconfig or our .idea/codeStyles. There will also be a lot of conflicts with the [pvbingo](https://github.com/NotEnoughUpdates/NotEnoughUpdates/tree/pvbingo/) branch, so theres a good chance lulo or nopo will ask you to rebase on that.

**Comment by walkerselby** [comment]
> This PR looks fine in general, but there are a LOT of problems with formatting, please see our .editorconfig or our .idea/codeStyles. There will also be a lot of conflicts with the [pvbingo](https://github.com/NotEnoughUpdates/NotEnoughUpdates/tree/pvbingo/) branch, so theres a good chance lulo or nopo will ask you to rebase on that.

Rebased and edited the formatting. git was being git, so I think it went correctly, but we shall see. Thank you for the review, and apologies for the formatting issues.

**Review by NopoTheGamer — APPROVED** [review]

**Review by walkerselby — COMMENTED** [review]
Should all be addressed and current with the master branch now.

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by romangraef — APPROVED** [review]
this is one fucked up git history, but at least the code is good :+1: maybe githubs squash + merge can make this work without looking to weird in git logs, im not sure what git operations they actually do with squash + merge

**Review by IRONM00N — APPROVED** [review]

**Pull request merged**